src/common/src/block_builder.cc

```diff
@@ -99,6 +111,28 @@ types::Block BlockBuilder::ConvertRecordToBlock(const std::vector<std::string> &
     return block;
 }
 
+types::Block BlockBuilder::ConvertRecordToBlockCTC(const std::vector<std::string> &record)
+{
+    types::Block block;
+    block.section     = record[1][0];
+    block.block       = std::stoi(record[BLOCK_BUILDER_CSV_FIELD_BLOCK_NUMBER]);
+    block.length      = std::stod(record[BLOCK_BUILDER_CSV_FIELD_BLOCK_LENGTH]);
+    block.grade       = std::stod(record[BLOCK_BUILDER_CSV_FIELD_BLOCK_GRADE]);
+    block.speed_limit = std::stod(record[BLOCK_BUILDER_SCHEDULE_FIELD_SPEED_LIMIT]);
+    AssignBlockInfrastructure(block, record[BLOCK_BUILDER_CSV_FIELD_INFRASTRUCTURE]);
+
+    if (block.has_station)
+    {
+        double t = std::stod(record[BLOCK_BUILDER_SCHEDULE_FIELD_TOTAL_TIME_TO_STATION]);
+        t = t * BLOCK_BUILDER_MINUTE_TO_SECOND_CONVERSION_FACTOR * BLOCK_BUILDER_SECOND_TO_MILISECONDS_CONVERSION_FACTOR;
```

Comment:
Better yet, why not make a new `convert` function

Reply:
Added `ConvertMinuteToMiliseconds()`